react/features/analytics/actions.js

```diff
@@ -0,0 +1,119 @@
+ * NOTE: Has to be used after JitsiMeetJS.init. Otherwise analytics will be
+ * null.
+ *
+ * @param {Store} store - The redux store in which the specified {@code action}
+ * is being dispatched.
+ * @returns {void}
+ */
+export function initAnalytics({ getState }) {
+    window.analyticsHandlers = [];
+
+    const { analytics } = JitsiMeetJS;
+    const { enabled, scriptURLs } = getState()['features/analytics'];
+    const config = getState()['features/base/config'];
+
+    if (!enabled || !analytics) {
+        return;
+    }
+
+    const machineId = JitsiMeetJS.getMachineId();
+    const handlerConstructorOptions = {
+        product: 'lib-jitsi-meet',
+        version: JitsiMeetJS.version,
+        session: machineId,
+        user: `uid-${machineId}`
+    };
+
+    _loadHandlers(scriptURLs, handlerConstructorOptions).then(
+        handlers => {
+            const permanentProperties = {
+                roomName: getState()['features/base/conference'].room,
+                userAgent: navigator.userAgent
+            };
+
+            const { group, server } = getState()['features/jwt'];
+
+            if (server) {
+                permanentProperties.server = server;
+            }
+            if (group) {
+                permanentProperties.group = group;
+            }
+
+            // optionally include local deployment information based on
+            // the contents of window.config.deploymentInfo
+            if (config.deploymentInfo) {
+                for (const key in config.deploymentInfo) {
+                    if (config.deploymentInfo.hasOwnProperty(key)) {
+                        permanentProperties[key]
+                            = config.deploymentInfo[key];
+                    }
+                }
+            }
+
+            analytics.addPermanentProperties(permanentProperties);
+            analytics.setAnalyticsHandlers(handlers);
+        },
+        error => analytics.dispose() && logger.error(error));
+}
+
+/**
+ * Tries to load the scripts for the analytics handlers.
+ *
+ * @param {Array} scriptURLs - The array of script urls to load.
+ * @param {Object} handlerConstructorOptions - the default options to pass when
+ * creating handlers.
+ * @returns {Promise} resolves with the handlers that have been
+ * successfully loaded and rejects if there are no handlers loaded or the
+ * analytics is disabled.
+ * @private
+ */
+function _loadHandlers(scriptURLs, handlerConstructorOptions) {
+    const handlersPromises = [];
+
+    scriptURLs.forEach(url =>
+        handlersPromises.push(
+            loadScript(url).then(
+                () => {
+                    return { type: 'success' };
+                },
+                error => {
+                    return {
+                        type: 'error',
+                        error,
+                        url
+                    };
+                }))
+    );
+
+    return new Promise((resolve, reject) => {
```

Comment:
I think you can save yourself of the wrapping promise and just `return Promise.all(...).then` then inside throw / return as you see fit.

Reply:
I will check it, I'm currently debugging something around it about an exception on mobile while loading google analytics.

Reply:
this